influxdb3_write/src/write_buffer/table_buffer.rs

```diff
@@ -183,20 +193,39 @@ impl TableBuffer {
         }
     }
 
-    pub fn record_batches(&self, schema: &Schema) -> Vec<RecordBatch> {
-        // ensure the order of the columns matches their order in the Arrow schema definition
-        let mut cols = Vec::with_capacity(self.data.len());
-        let schema = schema.as_arrow();
-        for f in &schema.fields {
-            cols.push(
-                self.data
-                    .get(f.name())
-                    .unwrap_or_else(|| panic!("missing field in table buffer: {}", f.name()))
-                    .as_arrow(),
-            );
+    pub fn record_batches(
+        &self,
+        schema: SchemaRef,
+        filter: &[Expr],
+    ) -> Result<Vec<RecordBatch>, anyhow::Error> {
```

Comment:
We only ever return one RecordBatch. Why return a Vec here?

Reply:
In theory it was going to potentially return many as other things were added to table buffer like splitting up the writes based on time. I can change it to a single and then add it back in later if that's how things progress?

Reply:
I think that makes more sense. Better to do what we have now and change the type sig later if need be.